dev/bots/test.dart

```diff
@@ -1413,56 +1417,89 @@ Future<void> _runFlutterTest(String workingDirectory, {
-      outputMode: outputMode,
-      skip: skip,
-      environment: environment,
-    );
+    if (!shouldProcessOutput) {
+      final OutputMode outputMode = outputChecker == null && printOutput
+        ? OutputMode.print
+        : OutputMode.capture;
 
-    if (outputChecker != null) {
-      final String message = outputChecker(result);
-      if (message != null)
-        exitWithError(<String>[message]);
+      final CommandResult result = await runCommand(
+        flutter,
+        args,
+        workingDirectory: workingDirectory,
+        expectNonZeroExit: expectFailure,
+        expectNoTests: runUnshuffled,
+        outputMode: outputMode,
+        skip: skip,
+        environment: environment,
+      );
+      if (outputChecker != null) {
+        final String message = outputChecker(result);
+        if (message != null)
+          exitWithError(<String>[message]);
+      }
+      return;
     }
-    return;
-  }
 
-  if (useFlutterTestFormatter) {
-    final FlutterCompactFormatter formatter = FlutterCompactFormatter();
-    Stream<String> testOutput;
-    try {
-      testOutput = runAndGetStdout(
+    if (useFlutterTestFormatter) {
+      final FlutterCompactFormatter formatter = FlutterCompactFormatter();
+      Stream<String> testOutput;
+      try {
+        testOutput = runAndGetStdout(
+          flutter,
+          args,
+          workingDirectory: workingDirectory,
+          expectNonZeroExit: expectFailure,
+          expectNoTests: runUnshuffled,
+          environment: environment,
+        );
+      } finally {
+        formatter.finish();
+      }
+      await _processTestOutput(formatter, testOutput);
+    } else {
+      await runCommand(
         flutter,
         args,
         workingDirectory: workingDirectory,
         expectNonZeroExit: expectFailure,
-        environment: environment,
+        expectNoTests: runUnshuffled,
       );
-    } finally {
-      formatter.finish();
     }
-    await _processTestOutput(formatter, testOutput);
+  }
+
+  if (shuffleOrder) {
+    final DateTime now = DateTime.now();
+    // Generates YYYYMMDD as the seed, so that testing continues to fail on the
+    // day it was originally run, and on other days the seed can be used to
+    // replicate failures.
+    final int seed = now.year * 10000 + now.month * 100 + now.day;
+    // Runs the tests not tagged with "no-shuffle" in a shuffled order.
+    print('Shuffling test order using arbitrary daily seed $seed.');
+    print('To run locally, run both "flutter test -x no-shuffle --test-randomize-ordering-seed=$seed" '
+      'and "flutter test -t no-shuffle" to run all tests (the latter runs all the tests that '
+      "currently don't work when shuffled, the former skips those tests and shuffles the order).");
+    await runTestsOnce(separateShuffled: true, seed: seed);
+    // Runs only tests that can't currently be shuffled (those tagged with
+    // "no-shuffle").
+    await runTestsOnce(separateShuffled: true);
```

Comment:
Is there anyway that the `runTestsOnce` with a seed can also run the no-shuffle tests as well as the shuffled? It seems counter intuitive that we need to call this twice and that each invocation runs different sets of tests.

Reply:
I thought about it, but for example, I'd have to duplicate the bodies of each of the conditional clauses in `runTestsOnce` and run the commands twice with different parameters in each clause, and the code ends up being a lot harder to follow than just calling this function twice. It's twice as much code because it really is just running the command twice, each time with different parameters.

Reply:
Ah got it. Would having a couple of different wrappers for this function help here? Perhaps `runShuffled`, `runNonShuffled`, `runSequential`? It is just a little confusing what the parameters are actually doing with this function.

Reply:
OK, I didn't make three more functions, but I reorganized this all a bit, so that the function takes an enum, which is one of `{shuffled, unshuffled, sequential}`, and the seed is calculated internally if the mode is `shuffled`. It should be pretty clear what's happening now.

Reply:
Nice. That is much easier to follow. Thx.